pkg/cmd/pkg/push.go

```diff
@@ -0,0 +1,176 @@
+package pkg
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/buildkite/cli/v3/pkg/cmd/factory"
+	"github.com/buildkite/go-buildkite/v3/buildkite"
+	"github.com/kr/pretty"
+	"github.com/oleiade/reflections"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+type newPackageConfig struct {
+	RegistrySlug string `flag:"registry"`
+	FilePath     string `flag:"file"`
+	FileName     string `flag:"file-name"`
+}
+
+var isStdInReadableFunc = isStdinReadable
+
+func isStdinReadable() (bool, error) {
+	stat, err := os.Stdin.Stat()
+	if err != nil {
+		return false, fmt.Errorf("failed to stat stdin: %w", err)
+	}
+
+	readable := (stat.Mode() & os.ModeCharDevice) == 0
+	return readable, nil
+}
+
+func NewCmdPackagePush(f *factory.Factory) *cobra.Command {
+	cmd := cobra.Command{
+		Use:   "push --registry <registry> {--file <file> | --file-name <filename> -}",
+		Short: "Push a new package to Buildkite Packages",
+		Example: heredoc.Doc(`
+			$ bk package push --registry my-registry --file my-package.tar.gz
+			$ cat my-package.tar.gz | bk package push --registry my-registry --file-name my-package.tar.gz - # Pass package via stdin, note hyphen as the argument
+		`),
+		Args:      cobra.MaximumNArgs(1),
+		ValidArgs: []string{"-"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cfg, err := populateFlags[newPackageConfig](cmd.PersistentFlags())
```

Comment:
i have a feeling the way flags are handled by other commands is much simpler - just giving you a primitive type to work with

Reply:
yeah, i originally handled it that way but got frustrated with pointer dereferencing and nil checking, so decided to replace it with the much more consistent and easy to read logic of (checks notes) reflection.

in retrospect, i've probably overcomplicated things here - happy to switch back to the other way ;)

Reply:
Haha nice. I dont think we've used actual pointer variables but just passed them into the flags by reference. Not sure if that works for this though 🤷‍♂️